.github/CONTRIBUTING.md

```diff
@@ -180,12 +182,15 @@ In case you've got a small change in most of the cases, your pull request would
 -   Write tests
 -   Follow the existing coding style
 -   Write a [good commit message](http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html)
+-   Make sure your PR has an issue and if it doesn't, please create one. This would help discussion with the community, and polishing ideas in case of a new feature.
+-   Make sure your PR's description contains GitHub's special keyword references that automatically close the related issue when the PR is merged. ([More info](https://github.com/blog/1506-closing-issues-via-pull-requests))
+-   When you make very minor changes to a PR of yours (like for example fixing a failing Travis build or some small style corrections or minor changes requested by reviewers) make sure you squash your commits afterward so that you don't have an absurd number of commits for a very small fix. ([Learn how to squash here](https://davidwalsh.name/squash-commits-git))
```

Comment:
Usually us maintainers tend to squash the commits when we merge the PR.
Also, this suggestion should suggest the opposite: `When you have lot of commits in your PR, it's good practice to squash all your commits in one single commit"